### PR TITLE
Feature/ssr (not working)

### PR DESCRIPTION
**Pull request by fredpointzero**

### CODE REVIEW

File: com.unity.render-pipelines.core/CoreRP/ShaderLibrary/ImageBasedLighting.hlsl [hunk 1]
```diff
@@ -31,6 +31,12 @@ real PerceptualRoughnessToMipmapLevel(real perceptualRoughness)
     return PerceptualRoughnessToMipmapLevel(perceptualRoughness, UNITY_SPECCUBE_LOD_STEPS);
 }
 
+// Mapping for convolved Texture2D
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.core/CoreRP/ShaderLibrary/ImageBasedLighting.hlsl]: add comment that this is empiric to match perceptually the GGX version of cubemap

File: com.unity.render-pipelines.high-definition/HDRP/Camera/HDCamera.cs [hunk 2]
```diff
@@ -504,6 +504,7 @@ public void SetupGlobalParams(CommandBuffer cmd, float time, float lastTime, uin
             cmd.SetGlobalVector(HDShaderIDs._ScreenParams,              screenParams);
             cmd.SetGlobalVector(HDShaderIDs._TaaFrameRotation,          taaFrameRotation);
             cmd.SetGlobalVectorArray(HDShaderIDs._FrustumPlanes,        frustumPlaneEquations);
+            cmd.SetGlobalVector(HDShaderIDs._ViewMatrixZRow, new Vector4(viewMatrix.m20, viewMatrix.m21, viewMatrix.m22, viewMatrix.m23));
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/HDRP/Camera/HDCamera.cs]: add comment => VGPR

File: com.unity.render-pipelines.high-definition/HDRP/Lighting/Reflection/ScreenSpaceReflectionRenderer.cs [hunk 3]
```diff
@@ -0,0 +1,385 @@
...
+        {
+            var ssReflection = VolumeManager.instance.stack.GetComponent<ScreenSpaceReflection>()
+                    ?? ScreenSpaceReflection.@default;
+
+            var projectionModel     = (Lit.ProjectionModel)ssReflection.deferredProjectionModel.value;
+
+            if (projectionModel == Lit.ProjectionModel.HiZ)
+                RenderPassAllocateRays(hdCamera, cmd, debug, debugTextureHandle, ssReflection);
+
+            RenderPassCastRays(hdCamera, cmd, debug, debugTextureHandle, ssReflection);
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/HDRP/Lighting/Reflection/ScreenSpaceReflectionRenderer.cs]: inside if

File: com.unity.render-pipelines.high-definition/HDRP/RenderPipeline/HDRenderPipeline.cs [hunk 4]
```diff
@@ -210,6 +211,11 @@ public HDRenderPipeline(HDRenderPipelineAsset asset)
                     new TexturePadding(asset.renderPipelineResources.texturePaddingCS)
                     );
             m_BufferPyramid = new BufferPyramid(bufferPyramidProcessor);
+            m_ScreenSpaceReflectionRenderer = new ScreenSpaceReflectionRenderer(
+                new ScreenSpaceReflectionRenderer.Settings { MaxRayAllocation = 2560 * 1440, ResolutionMip = 0 }, // TODO: Expose this in HDRenderPipeline
```
[the remark this answers is not in the document]
Comment (sebastienlagarde) [reply com.unity.render-pipelines.high-definition/HDRP/RenderPipeline/HDRenderPipeline.cs]: expose in UI 4 de rez
resize auto comme RT handle
hdRP asset: SSR control 4

File: com.unity.render-pipelines.high-definition/HDRP/RenderPipeline/HDRenderPipeline.cs [hunk 5]
```diff
@@ -307,6 +313,9 @@ void InitializeRenderTextures()
                 m_VelocityBuffer = RTHandles.Alloc(Vector2.one, filterMode: FilterMode.Point, colorFormat: Builtin.GetVelocityBufferFormat(), sRGB: Builtin.GetVelocityBufferSRGBFlag(), enableMSAA: true, name: "Velocity");
             }
 
+            if (m_Asset.renderPipelineSettings.supportSSR)
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/HDRP/RenderPipeline/HDRenderPipeline.cs]: inside

File: com.unity.render-pipelines.high-definition/HDRP/RenderPipeline/HDRenderPipeline.cs [hunk 6]
```diff
@@ -548,7 +559,7 @@ void Resize(HDCamera hdCamera)
             m_CurrentHeight = hdCamera.actualHeight;
         }
 
-        public void PushGlobalParams(HDCamera hdCamera, CommandBuffer cmd, DiffusionProfileSettings sssParameters)
+        public void PushGlobalParams(HDCamera hdCamera, CommandBuffer cmd, FrameSettings frameSettings, DiffusionProfileSettings sssParameters)
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/HDRP/RenderPipeline/HDRenderPipeline.cs]: frameSettings -> hdCamera

Comment (sebastienlagarde) [reply com.unity.render-pipelines.high-definition/HDRP/RenderPipeline/HDRenderPipeline.cs]: cleanage

File: com.unity.render-pipelines.high-definition/HDRP/RenderPipeline/HDRenderPipeline.cs [hunk 7]
```diff
@@ -848,6 +858,9 @@ public override void Render(ScriptableRenderContext renderContext, Camera[] came
                         hdCamera = HDCamera.Create(camera, m_VolumetricLightingSystem);
                     }
 
+                    if (currentFrameSettings.enableSSR)
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/HDRP/RenderPipeline/HDRenderPipeline.cs]: inside

File: com.unity.render-pipelines.high-definition/HDRP/RenderPipeline/HDRenderPipeline.cs [hunk 8]
```diff
@@ -944,6 +957,16 @@ public override void Render(ScriptableRenderContext renderContext, Camera[] came
                     // In both forward and deferred, everything opaque should have been rendered at this point so we can safely copy the depth buffer for later processing.
                     CopyDepthBufferIfNeeded(cmd);
                     RenderDepthPyramid(hdCamera, cmd, renderContext, FullScreenDebugMode.DepthPyramid);
+                    if (currentFrameSettings.enableSSR)
+                    {
+                        m_ScreenSpaceReflectionRenderer.RenderSSR(
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/HDRP/RenderPipeline/HDRenderPipeline.cs]: inside

File: com.unity.render-pipelines.high-definition/HDRP/RenderPipeline/HDRenderPipeline.cs [hunk 9]
```diff
@@ -944,6 +957,16 @@ public override void Render(ScriptableRenderContext renderContext, Camera[] came
...
+                            hdCamera, 
+                            cmd, 
+                            m_CurrentDebugDisplaySettings.IsDebugDisplayEnabled(),
+                            m_DebugFullScreenTempBuffer
+                        );
+                        m_FullScreenDebugPushed = true;
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/HDRP/RenderPipeline/HDRenderPipeline.cs]: SetFullScreenDebugTexture(hdCamera, cmd, m_CameraColorBuffer, FullScreenDebugMode.SSR);

File: com.unity.render-pipelines.high-definition/HDRP/ShaderVariables.hlsl [hunk 10]
```diff
@@ -280,11 +280,15 @@ CBUFFER_END
 CBUFFER_START(UnityLightingParameters)
 // Buffer pyramid
 float4  _ColorPyramidSize;              // (x,y) = Actual Pixel Size, (z,w) = 1 / Actual Pixel Size
-float4  _DepthPyramidSize;              // (x,y) = Actual Pixel Size, (z,w) = 1 / Actual Pixel Size
-float4  _CameraMotionVectorsSize;       // (x,y) = Actual Pixel Size, (z,w) = 1 / Actual Pixel Size
 float4  _ColorPyramidScale;             // (x,y) = Screen Scale, z = lod count, w = unused
+float4  _DepthPyramidSize;              // (x,y) = Actual Pixel Size, (z,w) = 1 / Actual Pixel Size
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/HDRP/ShaderVariables.hlsl]: clean pass?

**Comment by EvgeniiG** [comment]
Made obsolete by https://github.com/Unity-Technologies/ScriptableRenderPipeline/pull/1927.

**Pull request closed without merging**